src/Agent/NewRelic/Agent/Core/BrowserMonitoring/BrowserMonitoringWriter.cs

```diff
@@ -68,9 +68,18 @@ public string WriteScriptHeaders(string content)
             // Check if we found the tags and based on which comes last AND that this happens INSIDE the HEAD tag - do a replace on that.
             if ((xUaCompatibleFilterMatch.Success || charsetFilterMatch.Success) && (xUaCompatibleFilterMatch.Index < closingHeadTagIndex || charsetFilterMatch.Index > closingHeadTagIndex))
             {
-                var contentSubString = xUaCompatibleFilterMatch.Index > charsetFilterMatch.Index ? content.Substring(xUaCompatibleFilterMatch.Index, xUaCompatibleFilterMatch.Length) : content.Substring(charsetFilterMatch.Index, charsetFilterMatch.Length);
+                var match = charsetFilterMatch;
+                if(xUaCompatibleFilterMatch.Index > charsetFilterMatch.Index)
+                {
+                    match = xUaCompatibleFilterMatch;
+                }
+
+                var contentSubString = content.Substring(match.Index, match.Length);
                 var jsScriptWithContentSubString = string.Format("{0}{1}", contentSubString, _getJsScript());
-                return xUaCompatibleFilterMatch.Index > charsetFilterMatch.Index ? XUaCompatibleFilter.Replace(content, jsScriptWithContentSubString, 1) : CharsetFilter.Replace(content, jsScriptWithContentSubString, 1);
+
+                return content.Remove(match.Index, match.Length).Insert(match.Index, jsScriptWithContentSubString);
```

Comment:
I'm guessing it would be too risky to do `content.Replace(contentSubString, jsScriptWithContentSubString)`? It seems like that would be a little cleaner. You could also use a StringBuilder to do a string replace within a given index range in a single call. This will certainly work fine as-is, though.

Reply:
yea, i try to reduce the risk as much as I can.